### PR TITLE
WIP: Add Miracle Grue Configurator

### PR DESCRIPTION
**Pull request by skalnik**
While wanting to add support for rafts, I saw that we could easily just generate the config for Miracle Grue on the fly with values passed in. This is still a WIP. I'd like to see any settings easily adjustable by just including them in the POST data.
- [x] Tests for MiracleGrueConfigurator
- [x] Support any configuration data being POSTed in
- [x] Update README
- [x] Update Hubot script

### CODE REVIEW

**Comment by skalnik** [comment]
So what do we want to happen when someone specified quality and lineHeight?

And how do we want the API to look in regards to the "quality" option?

Current:

```
{
  "url": ["http://www.thingiverse.com/download:48876"],
  "count": 1,
  "scale": 1.0,
  "quality": "low",
  "density": 0.05,
  "config": "default"
}
```

Quality outside of args?:

```
{
  "url": ["http://www.thingiverse.com/download:48876"],
  "count": 1,
  "scale": 1.0,
  "quality": "low",
  "slicer_args": { "density": 0.05 }
}
```

Quality inside of args?:

```
{
  "url": ["http://www.thingiverse.com/download:48876"],
  "count": 1,
  "scale": 1.0,
  "slicer_args": { "quality": "low", "density": 0.05 }
}
```

**Comment by sshirokov** [comment]
I think "quality" should stay as it is now, since it's more abstract than a slicer option, and can evolve to imply more than just layer height. 

We can kill the connivence `density` arg entirely, since it didn't even work for a couple of minutes now anyway.

The rest, how about something more literal, so we can just merge it into the default config without thinking about it

``` json
{
  "url": ["http://www.thingiverse.com/download:48876"],
  "count": 1,
  "scale": 1.0,
  "quality": "low",
  "slicer_options": { "infillDensity": 0.05, "doRaft": true, "raftLayers": 2 }
}
```

**Comment by skalnik** [comment]
What happens when we specify lineHeight and quality. What takes precedence?

**Comment by sshirokov** [comment]
I'm thinking `slicer_options` overrides are final.

File: server/app.rb [hunk 1]
```diff
@@ -77,17 +79,32 @@ def progress
...
+                      0.34
+                    when 'high'
+                      0.1
+                    else
+                      0.27
+                    end
+
+      # Merge slicer_args into the quality array. Anything in slicer_args
+      # overwrites our notion of "quality"
+      slicer_args = {:lineHeight => quality}.merge(slicer_args)
```
Comment (sshirokov) [inline server/app.rb]: `:lineHeight => line_height` seems more correct here.

**Comment by sshirokov** [comment]
That was a bullshit failure. Grue fetch failed :rage4:

**Comment by sshirokov** [comment]
I edited the PR description 4U :dancers:

**Comment by skalnik** [comment]
Should be good to go, should ship with the Hubot script.

**Pull request merged**